kubeflow/trainer/rhai/instrumentation/traininghub_progression_test.py

```
@@ -0,0 +1,293 @@
+    )
+
+    algorithm_metadata = {
+        "name": "sft",
+        "metrics_file_pattern": "metrics_rank_*.jsonl",
+        "metrics_file_rank0": "metrics_rank_0.jsonl",
+    }
+
+    # Use a unique port to avoid conflicts
+    import random
+
+    test_port = random.randint(29000, 30000)
```

Ruff (0.15.12)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)